### PR TITLE
dev, v2.1, v3.0: add TiDB DM tutorial

### PR DESCRIPTION
**Pull request by CaitinChen**
### What is changed, added or deleted? 

Add TiDB DM tutorial

### What is the related PR or file link(s)? 

https://github.com/pingcap/docs/pull/1148
https://github.com/pingcap/docs/pull/1157
https://github.com/pingcap/docs/pull/1172
https://github.com/pingcap/docs/pull/1305

### Which version does your change affect? 

dev, v2.1, v3.0

**Note:** I'll update v3.0 and v2.1 files after dev files are fixed.

### Checklist 

- [x] Add a new file to `TOC.md`
- [x] No Tab spaces anywhere 
- [x] Leave a blank line both before and after a code block
- [x] Keep the first level heading consistent with `title` in metadata
- [x] Use *four* spaces for each level of indentation except that in `TOC.md`

### CODE REVIEW

File: dev/how-to/get-started/data-migration.md [hunk 1]
```diff
@@ -0,0 +1,722 @@
...
+## 结论
+
+本教程完成了 2 个练习：
+
+- 上游 3 个 MySQL Server 实例的分片迁移，每个实例的自增 ID 都互不重叠。
+- 上游 3 个 MySQL Server 实例的分片迁移，这些实例的自增 ID 存在冲突。
+
+本教程介绍了分片迁移中，DM 如何在集群中导入初始数据，以及 DM 读取 binlog，从而使下游 TiDB 集群与上游实例保持同步。
+
+关于 DM 的更多详情，请参考 [Data Migration 简介](/reference/tools/data-migration/overview.md)，或加入 [TiDB Community Slack](https://pingcap.com/tidbslack/) channel 参与讨论。
```
Comment (csuzhangxc) [inline dev/how-to/get-started/data-migration.md]: The link is broken.

Comment (CaitinChen) [reply dev/how-to/get-started/data-migration.md]: Thanks. But I find these links available, and this PR has passed the link check CI.

**Comment by CaitinChen** [comment]
@csuzhangxc Thanks a lot. PTAL again.

File: dev/how-to/get-started/data-migration.md [hunk 2]
```diff
@@ -0,0 +1,722 @@
...
+DM 功能如下：
+
+- 数据迁移
+    - 支持数据初次迁移任务的创建及导出，并在数据迁移过程中读取、应用来自源数据存储的 binlog，从而保持数据的同步。
+    - 通过合并上游的多个 MySQL 或 MariaDB 实例或集群的表，DM 能迁移生产环境中的分库分表。
+- 将 TiDB 作为 MySQL 或 MariaDB 的从库时，DM 能持续提高数据库水平扩展的能力，或在无需 ETL 作业的情况下，在 TiDB 上进行数据实时分析。
+
+本教程主要介绍如何使用 DM 迁移上游多个 MySQL 实例的一个分片表。包括两种场景：
+
+- 合并若干个互不冲突的表或分片，即这些表或分片的表结构并不会造成唯一键值对的冲突；
```
[the remark this answers is not in the document]
Comment (csuzhangxc) [reply dev/how-to/get-started/data-migration.md]: @CaitinChen does not update this line (L19)?

[the remark this answers is not in the document]
Comment (CaitinChen) [reply dev/how-to/get-started/data-migration.md]: @csuzhangxc Thanks for noticing this. Updated now.

File: dev/how-to/get-started/data-migration.md [hunk 3]
```diff
@@ -0,0 +1,722 @@
...
+    ```bash
+    for i in 1 2 3
+    do
+        echo  "mysql$i"
+        mysqld --defaults-group-suffix="$i" --initialize-insecure
+        mysqld --defaults-group-suffix="$i" &
+    done
+    ```
+
+5. （可选）执行 `jobs` 和/或 `pgrep -a mysqld` 以确保 MySQL Server 实例都在运行状态。
```
Comment (yikeke) [inline dev/how-to/get-started/data-migration.md]:
```suggestion
5. 执行 `jobs` 和/或 `pgrep -a mysqld` 以确保 MySQL Server 实例都在运行状态。
```
I don't see `(optional)` in https://pingcap.com/docs/v3.0/how-to/get-started/data-migration/? Is `（可选）` intended?

Comment (CaitinChen) [reply dev/how-to/get-started/data-migration.md]: Because it's not mandatory and users can skip this step. Of course, we can remove "可选."

**Comment by CaitinChen** [comment]
@yikeke Thanks.

**Pull request merged**